routing/index_graph.hpp

```diff
@@ -17,16 +18,16 @@
 
 namespace routing
 {
-class JointEdge final
+class SegmentEdge final
 {
 public:
-  JointEdge(Joint::Id target, double weight) : m_target(target), m_weight(weight) {}
-  Joint::Id GetTarget() const { return m_target; }
+  SegmentEdge(Segment const & target, double weight) : m_target(target), m_weight(weight) {}
```

Comment:
This class depends on Segment only, why not to move it out to an individual hpp?